react/src/pages/InfoPage.js

```diff
@@ -35,16 +37,24 @@ const InfoPage = () => {
         fetch(Server.Url("/__clearcache__", false));
     }
 
-    const InfoBox = ({title, children}) => {
+    const InfoBox = ({title, show = true, children}) => {
+        const [ showing, setShowing ] = useState(show);
         return <>
             <div className="container">
-                <b>{title}</b>
-                { title === "Versions" && <b className="tool-tip" data-text="Click to refresh." style={{float:"right",cursor:"pointer"}} onClick={info.refresh}>{Char.Refresh}&nbsp;</b> }
-                <ul className="top-level-list">
-                    <div className="info boxstyle" style={{paddingLeft:"8pt",paddingTop:"6pt",paddingBottom:"8pt"}}>
-                        {children}
+                { showing ? <>
+                    <b onClick={() => setShowing(false)} style={{cursor:"pointer"}}>{title}</b> &nbsp;<span onClick={() => setShowing(false)} style={{cursor:"pointer"}}>{Char.DownArrow}</span>
+                    { title === "Versions" && <b className="tool-tip" data-text="Click to refresh." style={{float:"right",cursor:"pointer"}} onClick={info.refresh}>{Char.Refresh}&nbsp;</b> }
+                    <ul className="top-level-list">
+                        <div className="box" style={{paddingLeft:"8pt",paddingTop:"6pt",paddingBottom:"8pt"}}>
+                            {children}
+                        </div>
+                    </ul>
+                </>:<>
+                    <b onClick={() => setShowing(true)} style={{cursor:"pointer"}}>{title}</b> &nbsp;<span onClick={() => setShowing(true)} style={{cursor:"pointer"}}>{Char.UpArrow}</span>
+                    <div className="box">
+                        Click <span onClick={() => setShowing(true)} style={{cursor:"pointer"}}><b>here</b></span> to <span onClick={() => setShowing(true)} style={{cursor:"pointer"}}>show</span>.
```

Comment:
Should the bold on something like this be explicit or part of the style of the span? Might be you do this elsewhere, too, I just happened to notice it and figured it was worth asking about. Building it in explicitly makes it harder to manipulate by styles.

Reply:
Yeah, lots of CSS common-ization needed. Will being doing gradually over time.